queue/src/main/java/com/bazaarvoice/emodb/queue/core/kafka/KafkaAdminService.java

```diff
@@ -0,0 +1,60 @@
+package com.bazaarvoice.emodb.queue.core.kafka;
+
+import org.apache.kafka.clients.admin.AdminClient;
+import org.apache.kafka.clients.admin.NewTopic;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collections;
+
+public class KafkaAdminService {
+    private static final Logger _log = LoggerFactory.getLogger(KafkaAdminService.class);
+    private final AdminClient adminClient;
+
+    public KafkaAdminService() {
+        this.adminClient = AdminClient.create(KafkaConfig.getAdminProps());
+    }
+
+    /**
+     * Creates a new Kafka topic with the specified configurations.
+     *
+     * @param topic             The name of the topic.
+     * @param numPartitions     Number of partitions.
+     * @param replicationFactor Replication factor.
+     */
+    public Boolean createTopicIfNotExists(String topic, int numPartitions, short replicationFactor, String queueType) {
+        Boolean isExisting =isTopicExists(topic);
+        if (! isExisting) {
+            //create the topic now
+            NewTopic newTopic = new NewTopic(topic, numPartitions, replicationFactor);
+            try {
+                adminClient.createTopics(Collections.singleton(newTopic)).all().get();
+                _log.info("Created topic: {} with numPartitions: {} ", topic, numPartitions, replicationFactor);
+            } catch (Exception e) {
+                 _log.error("Error creating topic {}: {}", topic, e.getMessage());
```

Comment:
todo: Do log the stacktrace of the exception for better debugging

Reply:
logged the stacktrace https://github.com/bazaarvoice/emodb/pull/838